spec/requests/tags_controller_spec.rb

```diff
@@ -137,6 +137,96 @@
     context "with tags_listed_by_group disabled" do
       before { SiteSetting.tags_listed_by_group = false }
       include_examples "successfully retrieve tags with topic_count > 0"
+
+      it "returns the right tags and categories tags for admin user" do
+        category.update!(tags: [test_tag])
+
+        sign_in(admin)
+
+        get "/tags.json"
+
+        expect(response.status).to eq(200)
+
+        tags = response.parsed_body["tags"]
+
+        expect(tags.length).to eq(2)
+
+        expect(tags[0]["name"]).to eq(test_tag.name)
+        expect(tags[0]["text"]).to eq(test_tag.name)
+        expect(tags[0]["description"]).to eq(test_tag.description)
+        expect(tags[0]["count"]).to eq(0)
+        expect(tags[0]["pm_count"]).to eq(0)
+        expect(tags[0]["target_tag"]).to eq(nil)
+
+        expect(tags[1]["name"]).to eq(topic_tag.name)
+
+        categories = response.parsed_body["extras"]["categories"]
+
+        expect(categories[0]["id"]).to eq(category.id)
+        expect(categories[0]["tags"][0]["name"]).to eq(test_tag.name)
```

Comment:
Should we check the number of tags returned here? in `categories[0]["tags"].count`